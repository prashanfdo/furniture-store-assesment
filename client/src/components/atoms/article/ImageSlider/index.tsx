import React from "react";
// Import Swiper React components
// import { Navigation, Pagination } from "swiper";
// import { Swiper, SwiperSlide } from "swiper/react";

// // Import Swiper styles
// import "swiper/css";
// import "swiper/css/navigation";
// import "swiper/css/pagination";
import { Article } from "types";

// TODO: implement this component to use multiple images to show in ArticleThumb as a image slider
// TODO: Need to add tests
type ImageSliderProps = {
  productImages?: Article["images"];
};
const ImageSlider: React.FC<ImageSliderProps> = ({ productImages }: ImageSliderProps) => {
  return <div>Image Slider</div>;
  // return (
  //   <div className="w-full h-full">
  //     <Swiper
  //       slidesPerView={1}
  //       spaceBetween={0}
  //       autoplay={{
  //         delay: 2500,
  //         disableOnInteraction: false,
  //       }}
  //       loop={true}
  //       pagination={{
  //         clickable: true,
  //       }}
  //       modules={[Pagination, Navigation]}
  //       className="mySwiper"
  //     >
  //       {productImages?.map((image) => (
  //         <SwiperSlide>
  //           <img className="object-contain w-full h-96" src={image.path} alt="" loading="lazy" />
  //         </SwiperSlide>
  //       ))}
  //     </Swiper>
  //   </div>
  // );
};

export default ImageSlider;
