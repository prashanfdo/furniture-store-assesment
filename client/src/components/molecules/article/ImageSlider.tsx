import React from "react";
// Import Swiper React components
import { Navigation, Pagination } from "swiper";
import { Swiper, SwiperSlide } from "swiper/react";

// Import Swiper styles
import "swiper/css";
import "swiper/css/navigation";
import "swiper/css/pagination";
import { Article } from "types";
import styled from "@emotion/styled";

const Container = styled.div`
  display: block;
  width: 100%;
  height: 100%;
  & .swiper {
    width: 100%;
    height: 100%;
  }
`;
const Img = styled.img`
  object-fit: contain;
  height: 260px;
  width: 100%;
`;

type ImageSliderProps = {
  productImages?: Article["images"];
};
const ImageSlider: React.FC<ImageSliderProps> = ({ productImages }: ImageSliderProps) => {
  return (
    <Container>
      <Swiper
        slidesPerView={1}
        spaceBetween={0}
        autoplay={{
          delay: 2500,
          disableOnInteraction: false,
        }}
        loop={true}
        pagination={{
          clickable: true,
        }}
        modules={[Pagination, Navigation]}
        className="mySwiper">
        {productImages?.map((image) => (
          <SwiperSlide>
            <Img src={image.path} alt="" />
          </SwiperSlide>
        ))}
      </Swiper>
    </Container>
  );
};

export default ImageSlider;
