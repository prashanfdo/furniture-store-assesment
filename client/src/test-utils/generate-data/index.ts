export * from "./article";
export * from "./category";
export * from "./search";
