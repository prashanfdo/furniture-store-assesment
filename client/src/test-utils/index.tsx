import { render, RenderOptions } from "@testing-library/react";
import fs from "fs";
import React, { FC, ReactElement } from "react";

import { BrowserRouter } from "react-router-dom";
import user from "@testing-library/user-event";

const wrapper: FC<{ children: React.ReactNode }> = ({ children }) => {
  return <BrowserRouter>{children}</BrowserRouter>;
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, "wrapper">) => {
  const view = render(ui, { wrapper, ...options });

  const style = document.createElement("style");
  style.innerHTML = fs.readFileSync("src/index.out.css", "utf8");
  document.head.appendChild(style);

  return view;
};
export * from "@testing-library/react";

export { faker } from "@faker-js/faker";
export * from "./generate-data";

export { customRender as render };
export { user };
