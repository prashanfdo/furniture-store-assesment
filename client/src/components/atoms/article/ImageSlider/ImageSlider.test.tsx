import { render } from "test-utils";
import ImageSlider from "./index";

describe("ImageSlider", () => {
  it("should render", () => {
    const { container } = render(<ImageSlider />);
    expect(container).toMatchSnapshot();
  });
});
