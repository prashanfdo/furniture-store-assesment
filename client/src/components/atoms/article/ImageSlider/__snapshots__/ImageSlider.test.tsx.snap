// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageSlider should render 1`] = `
<div>
  <div>
    Image Slider
  </div>
</div>
`;
